src/main/java/io/lettuce/core/protocol/SharedLock.java

```diff
@@ -26,6 +26,8 @@ class SharedLock {
 
     private final Lock lock = new ReentrantLock();
 
+    private final ThreadLocal<Integer> sharedCnt = ThreadLocal.withInitial(() -> 0);
```

Comment:
`sharedCnt` is a bit confusing to me.

Could we name it something like `localWriters` or `threadWriters` to indicate that we are keeping track of the current thread's counter for writes?

Reply:
`threadWriters` is looks sensible for me. thanks.